src/docbuilder/rustwide_builder.rs

```diff
@@ -310,6 +310,20 @@ impl RustwideBuilder {
                 }
 
                 let has_examples = build.host_source_dir().join("examples").is_dir();
+                let metadata = res.cargo_metadata.root();
+                let is_library = match metadata.targets[0].kind.as_slice() {
```

Comment:
We do the same check in `add_package`, and I worry over time the two might diverge. Could you extract both of them into an utility function?

https://github.com/rust-lang/docs.rs/blob/ffe043c00e5016cdc010bb87b8a6e1378e0fd876/src/db/add_package.rs#L40-L43

Reply:
I admit to stealing that block from `add_package`, so i'll go ahead and make it a method on Packages.